manager/raftpicker/raftpicker.go

```diff
@@ -14,46 +15,51 @@ type picker struct {
 	addr string
 	raft AddrSelector
 	conn *grpc.Conn
-	cc   *grpc.ClientConn
 }
 
 // Init does initial processing for the Picker, e.g., initiate some connections.
 func (p *picker) Init(cc *grpc.ClientConn) error {
-	p.cc = cc
-	return nil
-}
-
-func (p *picker) initConn() error {
-	if p.conn == nil {
-		conn, err := grpc.NewConn(p.cc)
-		if err != nil {
-			return err
-		}
-		p.conn = conn
+	conn, err := grpc.NewConn(cc)
+	if err != nil {
+		return err
 	}
+	p.conn = conn
 	return nil
 }
 
-// Pick blocks until either a transport.ClientTransport is ready for the upcoming RPC
-// or some error happens.
-func (p *picker) Pick(ctx context.Context) (transport.ClientTransport, error) {
-	p.mu.Lock()
-	if err := p.initConn(); err != nil {
-		p.mu.Unlock()
-		return nil, err
-	}
-	p.mu.Unlock()
-
+func (p *picker) updateConn() error {
 	addr, err := p.raft.LeaderAddr()
```

Comment:
May want to trace it to confirm, but this will cause two addr changes after a call to `PickAddr`.

Mind you, the Picker interface is pure garbage, so I have no clue if that is right or not.